doc/macro_utils_design.md

```diff
@@ -0,0 +1,240 @@
+macro utils design
+=======
+
+Macro utils is a set of C useful macros.
+
+This document explains how it was build and what every macro does. IN supporting understand rather than code accuracy, some macros will be chopped to deal with a reasonable number of arguments rather than their implementation that usually supports above 120 arguments.
```

Comment:
>IN [](start = 67, length = 2)

In

Reply:
In, in.

---
In reply to: [464137885](https://github.com/Azure/azure-macro-utils-c/pull/42#discussion_r464137885) [](ancestors = 464137885)